### PR TITLE
Add usage event to track when a iOS network device is used

### LINKED ISSUE #118479: Track iOS wireless debugging
We want to track how often iOS wireless debugging is used.

Relies on this PR: https://github.com/flutter/flutter/pull/118104

Will want to use Usage API in the tool to collect.

for example https://github.com/flutter/flutter/blob/2fd8250285492c7a4e216e2753f2bfd573d1b8f6/packages/flutter_tools/lib/src/ios/migrations/remove_framework_link_and_embedding_migration.dart#L93

### PR DESCRIPTION
**Pull request by vashworth**
We recently added wireless debugging for iOS network devices https://github.com/flutter/flutter/pull/118895 and want to track it's usage.

Fixes https://github.com/flutter/flutter/issues/118479.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by vashworth** [comment]
@jmagman You mentioned to me maybe using an existing tracking event such as when a user runs `flutter run` and adding data to it. I found this, which I believe is what you were referring to. 

https://github.com/flutter/flutter/blob/a07e8a6ac43d4582c86174592ebe45edc37d115a/packages/flutter_tools/lib/src/runner/flutter_command.dart#L1323-L1375

Unfortunately the handler is registered before the device is selected, though, which is why I decided to go with a new event.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart
Comment (jmagman) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:1545]: This usage would tell us an absolute number of wireless iOS devices, however I think it would be more interesting to know what percentage of iOS devices are wireless. How about something like:

```suggestion
    final Iterable<IOSDevice> iosDevices = devices.whereType<IOSDevice>();
    if (iosDevices.isNotEmpty) {
      final bool includesNetworkDevices =
          iosDevices.any((IOSDevice device) => device.interfaceType == IOSDeviceConnectionInterface.network);
      UsageEvent(
        'device',
        'ios-interface-type',
        label: includesNetworkDevices ? 'wireless' : 'usb',
        flutterUsage: globals.flutterUsage,
      ).send();
    }
```

**Comment by jmagman** [comment]
Spent some more time looking at this, maybe it's more appropriate to add this as a `RunCommand.usageValues()` `CustomDimensions` instead?

https://github.com/flutter/flutter/blob/224e6aa18cfd7ce5f839f8910d85f6cb5d4e455c/packages/flutter_tools/lib/src/commands/run.dart#L445

@christopherfujino can you weigh in on the best way to do this?  Or maybe @eliasyishak working on analytics right now may have a suggestion?

**Comment by christopherfujino** [comment]
> Spent some more time looking at this, maybe it's more appropriate to add this as a `RunCommand.usageValues()` `CustomDimensions` instead?
> 
> https://github.com/flutter/flutter/blob/224e6aa18cfd7ce5f839f8910d85f6cb5d4e455c/packages/flutter_tools/lib/src/commands/run.dart#L445
> 
> @christopherfujino can you weigh in on the best way to do this? Or maybe @eliasyishak working on analytics right now may have a suggestion?

I'll defer to Elias on this one :)

**Comment by eliasyishak** [comment]
@jmagman this is actually something I have been thinking about to standardize the usage events we send.

Does your `RunCommand.usageValues()` essentially resolve all the necessary data you want to send for collection?

**Comment by jmagman** [comment]
> Does your `RunCommand.usageValues()` essentially resolve all the necessary data you want to send for collection?

I guess that's my question, should @vashworth move her new usage dimension to that method?  It seems like that's the right place for it.

**Comment by eliasyishak** [comment]
I think that would make sense to add it to the `CustomDimensions` class that gets returned from the `RunCommand.usageValues` getter instead of sending it yourself through `UsageEvents.send(...)`. That way you can just piggy back on an event that is already being sent.

But do you need to add a new custom dimension?

**Comment by vashworth** [comment]
> But do you need to add a new custom dimension?

Yes, I believe so. I see that it would need to be added in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/reporting/custom_dimensions.dart. Looks like I also need to add it to Google Analytics (https://support.google.com/analytics/answer/2709829#zippy=%2Cin-this-article)?

**Comment by jmagman** [comment]
>  maybe it's more appropriate to add this as a `RunCommand.usageValues()` `CustomDimensions` instead?

Also `DriveCommand`, so the `RunCommandBase` commands.

**Review by jmagman — APPROVED** [review]
Thanks for doing this twice after I lead you astray the first time 🙂 
LGTM!  @eliasyishak should also review.

**Review by eliasyishak — APPROVED** [review]
LGTM! Let me know if you would like some help querying the database after this gets merged in as well

**Pull request merged**
